### ISSUE TITLE
Package request: Pop shell (Pop!_OS's tiling windows manager extension for GNOME shell)

### ISSUE THREAD

**Issue by operator-name** [NixOS/nixpkgs#92769]
**Project description**
> Pop Shell is a keyboard-driven layer for GNOME Shell which allows for quick and sensible navigation and management of windows. The core feature of Pop Shell is the addition of advanced tiling window management — a feature that has been highly-sought within our community. For many — ourselves included — i3wm has become the leading competitor to the GNOME desktop.

**Metadata**

* homepage URL: https://github.com/pop-os/shell
* source URL: https://github.com/pop-os/shell
* license: [gpl3](https://github.com/pop-os/shell/blob/master_focal/LICENSE)
* platforms: linux

**Comment by Technical27** [comment]
Here is what I use, anyone is free to go ahead and merge this (or fix it):
```nix
{ stdenv, fetchFromGitHub, nodejs, nodePackages, glib }:

stdenv.mkDerivation rec {
  pname = "gnome-shell-extension-popshell";
  version = "1.0.0";

  src = fetchFromGitHub {
    owner = "pop-os";
    repo = "shell";
    rev = "master_focal";
    sha256 = "1ygqzmvh6fjl1yc9rxb9mmis7ywaxqfnx6hn3clhl4y3vv5f19gn";
  };

  nativeBuildInputs = [ nodePackages.typescript glib ];

  makeFlags = [ "INSTALLBASE=$(out)/share/gnome-shell/extensions" ];
}
```

**Comment by jtojnar** [comment]
I am pretty sure I saw that somewhere but cannot find other than:

https://github.com/FelschR/nixos-config/blob/f56d52f32434904e3a95e29a761714e4ffd35e05/desktop/gnome.nix#L5-L21

**Comment by mogorman** [comment]
leaving this here while i work on it for myself. its running but by default a lot of its needed keys are clobbered. going to need some home-manager or nixos system dconf to work properly.
```nix
{ stdenv, fetchFromGitHub, nodejs, nodePackages, glib }:

stdenv.mkDerivation rec {
  pname = "gnome-shell-extension-pop-shell";
  version = "2020-08-13";

  src = fetchFromGitHub {
    owner = "pop-os";
    repo = "shell";
    rev = "55c40ce66a5a51e83c083063a704372e18ca49e5";
    sha256 = "1pv01jc85g1a3dbamcnvm9iwr03llxycnywx5p7ng04xyq5l6923";
  };

  uuid = "pop-shell@system76.com";

  nativeBuildInputs = [ glib nodePackages.typescript];

  makeFlags = [ "INSTALLBASE=$(out)/share/gnome-shell/extensions" ];

  postInstall = ''
    mkdir -p $out/share/gsettings-schemas/pop-shell-${version}/glib-2.0
    cp -R $out/share/gnome-shell/extensions/${uuid}/schemas \
          $out/share/gsettings-schemas/pop-shell-${version}/glib-2.0/schemas

    schemadir=${glib.makeSchemaPath "$out" "${pname}-${version}"}
    mkdir -p $schemadir
    cp -r $out/share/gnome-shell/extensions/$uuid/schemas/* $schemadir
  
  '';

  meta = with stdenv.lib; {
    description = "i3wm-like keyboard-driven layer for GNOME Shell";
    homepage = "https://github.com/pop-os/shell";
    license = licenses.gpl3;
    maintainers = with maintainers; [ mog ];
    platforms = platforms.linux;
  };
}
```

**Comment by mogorman** [comment]
hmm it doesnt seem to work with wayland so im not going to try it till they get that sorted

**Comment by ratsclub** [comment]
As I can tell there are two PRs[^1][^2] trying to package it. Which one should I give a try?

[^1]: https://github.com/NixOS/nixpkgs/pull/104160
[^2]: https://github.com/NixOS/nixpkgs/pull/102150

**Comment by jtojnar** [comment]
Neither will work with GNOME 40.

**Comment by ratsclub** [comment]
Isn't there a way to use the old GNOME 3.0 package?

**Comment by buckley310** [comment]
Theoretically yes, but I don't think it's easy.
Among other things, I believe the GNOME 3.38->40 update is linked to the GTK 3->4 update, which is fairly invasive.

**Comment by ratsclub** [comment]
Oh, I see. I'll then wait until October's update that might give GNOME 40 support!

**Comment by genofire** [comment]
Solved in  #147542 - i believe we could close this ISSUE.
Maybe you add yourself to maintainers ;)
